### DISCUSSION TITLE
Filters while app isn't in foreground

### DISCUSSION THREAD

**Discussion by brian7704** [pedroSG94/RootEncoder#1671] [Q&A]
I set my app up to stream while it's in the background, but now I'm adding filters and they only seem to work when the app is in the foreground. Is there a way to make them work when the app is in the background as well?

**Comment by pedroSG94** [comment]
Hello,

If you are using replaceView method to go to background and foreground then it is expected.
The problem is that replaceView, replace the OpenGl render used for filter removing all filters so you will need set filters again after that.
Other solution is use StreamBase classes because that class doesn't need change the OpenGl render to go to background.

**Comment by brian7704** [comment]
Thank you Pedro, I will try the StreamBase classes. One other thing I noticed is that there is no audio from the device's microphone when my app is in the background. Is there a way to fix that?

**Comment by brian7704** [comment]
I'm trying to switch to using GenericStream instead of GenericCamera2 but I don't see a way to detect when the camera changes. Is that possible to do in GenericStream?

**Comment by pedroSG94** [comment]
What do you mean with camera changes?
Can you tell me the method used previously that you want implement?

**Comment by brian7704** [comment]
I was previously using CameraCallbacks.onCameraChanged. However I was able to get it working by making my own switchCamera() method that calls Camera2Source.switchCamera() and also does what I need. Thanks for your help!
